defs/go.bzl

```diff
@@ -57,24 +62,26 @@ def _go_genrule_impl(ctx):
         # Set GOPATH and GOROOT to absolute paths so that commands can chdir without issue
         "export GOPATH=" + ctx.configuration.host_path_separator.join(["$$GO_GENRULE_EXECROOT/" + p for p in go_paths]),
         "export GOROOT=$$GO_GENRULE_EXECROOT/" + go.root,
+        "export PATH=$$GO_GENRULE_EXECROOT/" + go.root + "/bin:$$PATH",
         ctx.attr.cmd.strip(" \t\n\r"),
     ]
     resolved_inputs, argv, runfiles_manifests = ctx.resolve_command(
         command = "\n".join(cmd),
         attribute = "cmd",
         expand_locations = True,
-        make_variables = _compute_genrule_variables(all_srcs, depset(ctx.outputs.outs)),
+        make_variables = _compute_genrule_variables(
+            all_srcs,
+            depset(ctx.outputs.outs),
+            dep_import_paths,
+        ),
         tools = ctx.attr.tools,
         label_dict = label_dict,
     )
 
-    paths = [go.root + "/bin", "/bin", "/usr/bin"]
```

Comment:
hm, so this change is now breaking kubernetes for reasons I still don't fully understand:
- `ctx.configuration.default_shell_env` seems to just be an empty dict
- yet `PATH` is initially getting set to (just) `/usr/bin` (which I don't understand, unless bazel is adding the directory containing bazel to `PATH` somehow?)
- and one of our `go_genrules` depends on `cp`, which is in `/bin`.

can you maybe keep the explicit `/bin/` and `/usr/bin` paths as we had here initially, but still set the `go.root` path at execution-time?